### ISSUE TITLE
Mounting a sub-directory doesn't work when using pseudo path

### ISSUE THREAD

**Issue by Bog999** [nfs-ganesha/nfs-ganesha#744]
Hi, I'm trying to mount a sub-directory while pseudo paths are enabled and it fails, but when I try the same thing without pseudo paths it works. Is this intentional?

I'm testing this on `V4-dev.9`, also tested on `V3.3`

Configuration:
```
EXPORT_DEFAULTS
{
	Protocols = 3;
}

NFS_CORE_PARAM
{
	Enable_NLM = false;
	Enable_RQUOTA = false;
	Protocols = 3;
	mount_path_pseudo = true;
	Plugins_Dir = "./FSAL/FSAL_VFS/vfs";
	RPC_Ioq_ThrdMax = 1;
}

EXPORT
{
	Export_ID=100;
	Protocols = 3;
	Transports = TCP,UDP;
	Path = /opt/ganshare;
	Pseudo = /gs1;
	Attr_Expiration_Time = 0;
	FSAL {
		Name = VFS;
	}
	
	CLIENT
	{
		Clients = 127.0.0.1;
		Access_Type = "RW";
		Squash = No_Root_Squash;
	}
}
```
When I mount /gs1 it works fine, but when I try to mount /gs1/subdir I get access denied by server.
If I remove `mount_path_pseudo = true` mounting the sub-directory works.

I tried to find out what the problem is
`mnt_Mnt()`, this line `export = get_gsh_export_by_pseudo(arg->arg_mnt, false);` returns the export for `/gs1` and then `op_ctx->fsal_export->exp_ops.lookup_path(op_ctx->fsal_export, arg->arg_mnt,	&obj, NULL)))` is called and it calls `sub_export->exp_ops.lookup_path(sub_export, path, &sub_handle, &attrs)` which calls `open_dir_by_path_walk(-1, path, &stat)` and searches for the pseudo path like it was an absolute path on the server's file system.

Stack trace:
```
open_dir_by_path_walk commonlib.c:563
vfs_lookup_path handle.c:1838
mdcache_lookup_path mdcache_handle.c:1553
mnt_Mnt mnt_Mnt.c:155
nfs_rpc_process_request nfs_worker_thread.c:1481
```

**Comment by ffilz** [comment]
OK, I think I get what the issue is. I'll take a look at doing a patch.

**Comment by Bog999** [comment]
Thanks!
I think that the call to `op_ctx->fsal_export->exp_ops.lookup_path(op_ctx->fsal_export, arg->arg_mnt, &obj, NULL)))` should be with the real absolute path and not with the pseudo path.

**Comment by ffilz** [comment]
This has been fixed by commit 2acd535c80d523ec354e09ba787983053d2f5a43

Unfortunately that is not slated for back port to V3.x, perhaps we should change that since we don't have a close date for V4.0 development yet.

**Comment by dang** [comment]
Or maybe just release 4.0?  It's coming up on 2 years...
